### DISCUSSION TITLE
Quarkus equivalent to Spring DataSourceUtils.getConnection(datasource)

### DISCUSSION THREAD

**Discussion by RomainWilbert** [quarkusio/quarkus#31031] [Q&A] [answered]
Is it safe to assume that the Agroal implementation of datasource.getConnection() is equivalent to Spring DataSourceUtils.getConnection(datasource) ?

Thanks :)

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
It's similar, yes

**Comment by RomainWilbert** [comment]
`DataSourceUtils.getConnection()` returns the same instance of HikariProxyConnection while being in a transaction, whereas a new ConnectionWrapper instance is returned on each call of `Datasource.getConnection()`.
Is this normal behavior ? Can this affect performance on heavy loads ?
